CSharp/README.md

```diff
@@ -23,6 +24,115 @@ var httpClient = new HttpClient();
 IUserService userService = new UserService();
 ```
 
+### Don't use explicit *this* reference
+
+It is redundant and does not add for readability. For more information read [this SO post](http://stackoverflow.com/questions/5885249/to-use-or-not-to-use-the-this-qualifier-in-c-sharp).
+
+```csharp
+// Bad
+this.ValidateParameters();
+
+// Good
+ValidateParameters();
+```
+
+### Use *string* instead of *String* keyword
+
+```csharp
+// Bad
+String.IsNullOrEmpty(name);
+
+// Good
+string.IsNullOrEmpty(name);
+```
+
+### Use readonly fields when possible
+
+### Use string interpolation
+
+```csharp
+// Bad
+var message = string.Format("My name is {0} {1}.", person.FirstName, person.LastName);
+
+// Good
+var message = $"My name is {person.FirstName} {person.LastName}.");
+```
+
+## Layout
+
+### Use *four spaces* per indentation level
+
+```csharp
+// Bad
+public void SomeMethod()
+{
+  DoSomethingFirst();
+  DoSomerthingLater();
+}
+
+// Good
+public void SomeMethod()
+{
+    DoSomethingFirst();
+    DoSomerthingLater();
+}
+```
+
+### Remove unused or redudant namespaces
+
+### Use single line and lambda getters
+
+```csharp
+// Good
+public string Name { get; private set; }
+
+// Good
+public string UppercaseName => Name.toUpperCase(); 
+```
+
+### Don't use braces for just one line

```

Comment:
Some time ago there was a discussion about it, after research many other conventions we concluded that it was a good idea to always use braces, the reason is mainly related to version control tools, diffs are simpler when you add a line and it is less error prone on resolving conflicts. There were really persuasive examples.